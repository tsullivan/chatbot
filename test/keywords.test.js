const request = require('supertest');
const { runServer } = require('../server');
const { utilFactory } = require('./utils');

const app = runServer();
const agent = request.agent(app);
const { handshake } = utilFactory(agent);

describe('Keywords', () => {
  test('should give a requested joke', async () => {
    await handshake();

    const res = await agent.post('/chat').send({ format: 'user', message: 'joke 5' });
    expect(res.statusCode).toEqual(200);
    expect(res.body.message).toMatchSnapshot();
  });
});
