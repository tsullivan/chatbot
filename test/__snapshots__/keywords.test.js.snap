// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Keywords should give a requested joke 1`] = `
"Here is joke number 5:
Why did the man throw a butterfly out of the window?
Because he wanted it to be able to smell around the world!"
`;
